shipyard_blueprints/microsoft-power-bi/shipyard_microsoft_power_bi/microsoft_power_bi.py

```diff
@@ -0,0 +1,280 @@
+            request_id = response.headers.get("RequestId")
+            PowerBiUtils.wait_for_dataset_refresh_completion(
+                self, group_id, dataset_id, request_id, wait_time=wait_time
+            )
+        else:
+            return response
+
+    def refresh_dataflow(
+        self,
+        group_id: str,
+        dataflow_id: str,
+        wait_for_completion: bool,
+        wait_time: int = 60,
+    ):
+        """
+        Triggers a refresh job for the specified dataflow with the option to wait that job to complete.
+        resource: https://docs.microsoft.com/en-us/rest/api/power-bi/dataflows/refreshdataflow
+
+        @param group_id: Group/Workspace ID
+        @param dataflow_id: Dataflow ID
+        @param wait_for_completion: If True, waits for the refresh job to complete.
+        @param wait_time: If wait_for_completion is True, the number of seconds to wait for the refresh job to complete.
+        @return: response from the request or None if wait_for_completion is True.
+        """
+
+        data = {"refreshRequest": "ShipyardRefresh"}
+        response = self._request(
+            f"{BASE_URL}/groups/{group_id}/dataflows/{dataflow_id}/refreshes",
+            method="POST",
+            data=json.dumps(data),
+        )
+        if wait_for_completion:
+            PowerBiUtils.wait_for_dataflow_refresh_completion(
+                self, group_id, dataflow_id, wait_time=wait_time
+            )
+        else:
+            return response
+
+    def get_dataflow_transactions(self, group_id: str, dataflow_id: str):
+        """
+        Gets the transactions for the specified dataflow.
+        resource: https://docs.microsoft.com/en-us/rest/api/power-bi/dataflows/getdataflowtransactions
+
+        @param group_id: Group/Workspace ID
+        @param dataflow_id: Dataflow ID
+        @return: response from the request
+        """
+        return self._request(
+            f"{BASE_URL}/groups/{group_id}/dataflows/{dataflow_id}/transactions",
+            method="GET",
+        )
+
+    def check_recent_dataset_refresh_by_request_id(
+        self,
+        group_id: str,
+        dataset_id: str,
+        request_id: str,
+        number_of_refreshes: int = 10,
+    ):
+        """
+        Gets the status of the most recent refreshes for the specified dataset and checks if the specified request ID is
+        present in those refreshes.
+        resource: https://docs.microsoft.com/en-us/rest/api/power-bi/datasets/getdatasetrefreshes
+
+        @param group_id: Group/Workspace ID
+        @param dataset_id: Dataset ID
+        @param request_id: Request ID to check for. This is the ID returned by the refresh request.
+        @param number_of_refreshes: The number of recent refreshes to check.
+        @return: Response from the request.
+        """
+        response = self._request(
+            f"{BASE_URL}/groups/{group_id}/datasets/{dataset_id}/refreshes?$top={number_of_refreshes}",
```

Comment:
I've considered using $top=1 to fetch the latest run, similar to how we handle the dataflow check.

However, I have some reservations about this method. Primarily, it introduces a higher susceptibility to external factors, such as scheduled refreshes or interventions by other users. By relying on `$top=1`, we might inadvertently capture a run that wasn't initiated by our API call. This could lead to unintended consequences or confusion.  I didn't see the option to do this with dataflows but dataflows do not allow for multiple refreshes to be queued.